### PR TITLE
Fixed type annotations and added mypy as static type checker

### PR DESCRIPTION
**Pull request by BlankSpruce**
There are still places that would need some attention in terms of either proper type-checking - like unqualified container types - or some refactoring in general.

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[mypy-docopt]
+ignore_missing_imports = True
+
+[mypy-lark.*]
+ignore_missing_imports = True
+
+[mypy-pytest]
+ignore_missing_imports = True
```
[the remark this answers is not in the document]
Comment (BlankSpruce) [reply mypy.ini]: Indeed it is. Would you prefer file ending with newline?

[the remark this answers is not in the document]
Comment (Scony) [reply mypy.ini]: No, it's good. Was just wondering what is that sign github presented

**Comment by Scony** [comment]
I'll finish review tomorrow

**Comment by Scony** [comment]
thanks

**Pull request merged**